backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -175,7 +175,21 @@ const IR::Node* RemoveUnusedMetadataFields::preorder(IR::DpdkAsmProgram *p) {
                     IR::IndexedVector<IR::StructField> usedMetadataFields;
                     for (auto field : st->fields) {
                         if (used_fields.count(field->name.name)) {
-                            usedMetadataFields.push_back(field);
+                            // DPDK implements bool and error types as bit<8>
```

Comment:
These changes are made to get the correct offset of metadata fields. 
Earlier, bool and error to bit<8> conversion was happening directly while emitting assembly. Now added this while removing the unused metadata to get the right offset in the next pass.